lib/sidekiq/middleware/server/active_record.rb

```diff
@@ -2,6 +2,21 @@ module Sidekiq
   module Middleware
     module Server
       class ActiveRecord
+        def initialize
+          # We can't use this middleware with the Rails reloader.
+          #
+          # The reloader needs the active connection to clear its query cache
+          # before releasing it otherwise other workers will see dirty query
+          # cache data.
+          #
+          # We need to make sure folks remove this middleware if they've added
+          # it themselves.
+          #
+          if defined?(Sidekiq::Rails::Reloader) && Sidekiq.options[:reloader].is_a?(Sidekiq::Rails::Reloader)
```

Comment:
I did that initially but moved it here hoping that it might raise earlier in the Sidekiq boot, but sadly it only raises during a job run. Happy to move it back if it seems more sensible.